ha/core/controllers/pcs/node_controller.py

```diff
@@ -71,7 +71,28 @@ def stop(self, node_id: str, timeout: int, **op_kwargs) -> dict:
             ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        check_cluster = op_kwargs.get("check_cluster") if op_kwargs.get("check_cluster") is not None else True
+        # Get the node_name (pvtfqdn) fron nodeid and raise exception if node_id is not valid
+        node_name = self._get_node_name(node_id=node_id)
+        try:
+            timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
+            node_status = self._system_health.get_node_status(node_id=node_id).get("status")
+            if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
+                Log.info(f"For stop {node_name}, Node already in offline state.")
+                status = f"Node {node_name} is already in offline state."
```

Comment:
*{node_name}
Please use node_id since caller has passed node_id

Reply:
Please check and update all status returns for the comment above.

Reply:
Changed to node_id.